extension/content/components/recipes/form/filters/FallbackFO.js

```diff
@@ -0,0 +1,59 @@
+// @ts-nocheck
+import React from "react";
+import { Badge, ControlLabel, FormGroup } from "rsuite";
+
+import JsonEditor from "devtools/components/common/JsonEditor";
+import {
+  ACTION_UPDATE_DATA,
+  useRecipeDetailsData,
+  useRecipeDetailsDispatch,
+} from "devtools/contexts/recipeDetails";
+import { partitionFO } from "devtools/components/recipes/form/filters/partitionFO";
+
+export default function FallbackFO() {
+  const data = useRecipeDetailsData();
+  const dispatch = useRecipeDetailsDispatch();
+  const [invalidJSON, setInvalidJson] = React.useState(false);
+
+  let knownFO = [];
+  let additionalFO = [];
+  if (data.filter_object) {
+    [knownFO, additionalFO] = partitionFO(data.filter_object);
+  }
```

Comment:
i think if you just do the split here you never need to call `partitionFO` ever again so I'm not sure that even needs to be a helper function. you can simply reuse `knownFO` from this scope in `handleChange`.